### PR TITLE
Updated mixpanel to build itself with Carthage.

### PR DESCRIPTION
**Pull request by simX**
This allows Mixpanel to be built using Carthage: https://github.com/Carthage/Carthage .  Note: this change was originally created by GitHub user justinmakaila, but in my testing it seems to work fine.

### CODE REVIEW

File: Mixpanel/MPObjectSelector.h [hunk 1]
```diff
@@ -6,7 +6,7 @@
 //  Copyright (c) 2014 Mixpanel. All rights reserved.
 //
 
-#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
```
Comment (Quuxplusone) [inline Mixpanel/MPObjectSelector.h]: This header file doesn't use anything from UIKit; Foundation should be sufficient. (You might need to add UIKit somewhere else, though.)

**Comment by simX** [comment]
Updated the pull request to fix quuxplusone's comment, and to merge in the latest Mixpanel changes.

**Comment by simX** [comment]
Also, the automatic Travis CI build continually gives me an error here on GitHub:

The command "openssl aes-256-cbc -k "$ENCRYPTION_SECRET" -in scripts/profile/HelloMixpanel_Ad_Hoc.mobileprovision.enc -d -a -out scripts/profile/HelloMixpanel_Ad_Hoc.mobileprovision" failed and exited with 1 during .

Not sure what that's about.

**Comment by pcantrell** [comment]
+1 Works for me
Fixes #265

**Comment by simX** [comment]
Actually this PR has a bug in the way that the framework looks for xib files when presenting a survey that will cause it to crash.  I have an updated fork that will fix this problem.

**Comment by samgreen** [comment]
I really want to get Mixpanel compatible with Carthage! This is definitely a priority for us. 

That said, I think providing two separate `.xcodeproj` files in the same repo might be kind of confusing for users who simply clone master. I'm researching Carthage now to see if there's an alternative to this.

Do you guys have any thoughts as to making this process as simple and clear as possible for both CocoaPods and Carthage users?

**Comment by simX** [comment]
We can put the framework target in the existing HelloMixpanel.xcodeproj file, but it didn't seem appropriate because it was just sample code; there weren't any targets for building the static library.  Also, there are some quirks about finding the xcodeproj file that makes it easier to exist at top level.

**Comment by samgreen** [comment]
Okay, we're gonna go ahead and move forward with this PR. I've verified it locally and it seems to building well with Carthage. @simX Can you please address the `UIKit` import comment above, and ensure we're **not** importing `UIKit` anywhere that it is not needed? After that I'll feel comfortable merging these changes in to master.

**Comment by simX** [comment]
I've merged in the latest Mixpanel changes.  I've tidied up the project file a bit, fixed some build errors due to removal of the libMPCategoryHelpers.a static library, and fixed a build warning.  I've also reverted all the unnecessary imports of UIKit.

**Comment by simX** [comment]
Note: going forward, this new Xcode project file will need to be kept up-to-date with the latest file additions and removals as they happen (such as the removal of the libMPCategoryHelpers.a static library).  An xcodebuild or carthage build might need to be added to any battery of tests before each release.

**Comment by simX** [comment]
I looked in issue #265 and noticed @Rupert-RR's comment about changing references to the mainBundle.  A similar issue was fixed in #207.  I've noticed a number of other places where mainBundle is referenced.  Should we change those to explicitly use bundleForClass: instead of mainBundle as well?

**Comment by samgreen** [comment]
@simX The bundle stuff needs to be fixed everywhere. I thought we were using `bundleForClass:` everywhere in the code base. Let me know what you find.

**Comment by simX** [comment]
@samgreen Just took a look, it seems I was wrong.  The framework uses [NSBundle mainBundle] in some places, but it's appropriate usage.  In various places you're getting the CFBundleVersion, CFBundleShortVersionString, and CFBundleIdentifier from the main application for super-properties and people properties, as well as supported interface orientations and snapshot_config.json, which all seem to appropriately come from the main bundle.

I think this pull request is ready. :)

**Comment by samgreen** [comment]
@simX Thank you so much for this contribution. Merging now.

**Comment by alec-c4** [comment]
great, but could you please update readme?

**Pull request merged**